core/ops.rs

```diff
@@ -136,44 +108,61 @@ impl OpResult {
 #[serde(rename_all = "camelCase")]
 pub struct OpError {
   #[serde(rename = "$err_class_name")]
-  class_name: &'static str,
-  message: String,
-  code: Option<&'static str>,
+  pub class_name: &'static str,
+  pub message: String,
+  pub code: Option<&'static str>,
 }
 
-pub fn serialize_op_result<R: Serialize + 'static>(
+pub fn to_op_result<R: Serialize + 'static>(
+  state: &OpState,
   result: Result<R, Error>,
-  state: Rc<RefCell<OpState>>,
 ) -> OpResult {
   match result {
     Ok(v) => OpResult::Ok(v.into()),
     Err(err) => OpResult::Err(OpError {
-      class_name: (state.borrow().get_error_class_fn)(&err),
+      class_name: (state.get_error_class_fn)(&err),
       message: err.to_string(),
       code: crate::error_codes::get_error_code(&err),
     }),
   }
 }
 
+#[inline]
+pub fn serialize_op_result<'a, R: Serialize + 'static>(
+  scope: &mut v8::HandleScope<'a>,
+  state: &OpState,
+  result: Result<R, Error>,
+) -> Result<v8::Local<'a, v8::Value>, serde_v8::Error> {
+  match result {
+    Ok(v) => serde_v8::to_v8(scope, v),
+    Err(err) => serde_v8::to_v8(
+      scope,
+      OpError {
+        class_name: (state.get_error_class_fn)(&err),
+        message: err.to_string(),
+        code: crate::error_codes::get_error_code(&err),
+      },
+    ),
+  }
+}
+
 /// Maintains the resources and ops inside a JS runtime.
 pub struct OpState {
   pub resource_table: ResourceTable,
-  pub op_table: OpTable,
   pub get_error_class_fn: GetErrorClassFn,
-  pub(crate) tracker: OpsTracker,
+  pub tracker: OpsTracker,
```

Comment:
Why is `tracker` now public?

Reply:
OpsTracker is used by the macro generated code to call op_state.tracker.track_sync()